converter/service/converter.py

```diff
@@ -0,0 +1,215 @@
+param_charset = 'charset'
+default_charset = 'utf-8'
+
+json_key_packages = 'packages'
+param_url = 'url'
+url_path = '/transform'
+
+
+def run_server(server_class=HTTPServer):
+    """Runs a builtin python server using the given server_class.
+
+    :param server_class: server
+    :type server_class: HTTPServer
+    :return: None
+    """
+    server_address = (HOST_NAME, PORT_NUMBER)
+    httpd = server_class(server_address, Handler)
+    logger.warning('Server Starts on port - %s', PORT_NUMBER)
+    try:
+        httpd.serve_forever()
+    except KeyboardInterrupt:
+        httpd.server_close()
+        logger.warning('Server Stops on port - %s', PORT_NUMBER)
+
+
+class Handler(BaseHTTPRequestHandler):
+    def do_GET(s):
+        """
+        Respond to the GET request. The expected format of this request is:
+                http://<host>:<port>/transform?url=<url> with `User-Agent`
+                and `Accept` headers
+        """
+        errors = _validate_request(s)
+        if errors:
+            s.send_error(HTTPStatus.BAD_REQUEST, errors)
+            return
+
+        query = dict(parse_qsl(urlparse(s.path).query))
+        if param_url not in query:
+            s.send_error(HTTPStatus.BAD_REQUEST,
+                         ErrorResponse.PARAM_NOT_PRESENT.to_msg(param_url))
+            return
+
+        user_agent = s.headers.get(header_user_agent)
+        accept = s.headers.get(header_accept)
+        decoded_url = query.get(param_url)
+
+        try:
+            json_response = handle(decoded_url, user_agent, accept)
+            s.send_response(HTTPStatus.OK)
+            content_header = gen_universe.format_universe_repo_content_type(
+                _get_repo_version(accept))
+            s.send_header(header_content_type, content_header)
+            s.end_headers()
+            s.wfile.write(json_response.encode())
+        except Exception as e:
+            s.send_error(HTTPStatus.BAD_REQUEST, str(e))
+
+
+def handle(decoded_url, user_agent, accept):
+    """Results in either an error or the requested json data
+
+    :param decoded_url: The url to be fetched from
+    :type decoded_url: str
+    :param user_agent: User-Agent header value
+    :type user_agent: str
+    :param accept: Accept header value
+    :return Requested json data
+    :rtype str (a valid json object)
+    """
+    logger.debug('Url : %s\n\tUser-Agent : %s\n\tAccept : %s',
+                 decoded_url, user_agent, accept)
+    repo_version = _get_repo_version(accept)
+    dcos_version = _get_dcos_version(user_agent)
+    logger.debug('Version %s\nDC/OS %s', repo_version, dcos_version)
+
+    req = Request(decoded_url)
+    req.add_header(header_user_agent, user_agent)
+    req.add_header(header_accept, accept)
+    try:
+        res = urlopen(req, timeout=MAX_TIMEOUT)
+        charset = res.info().get_param(param_charset) or default_charset
+        raw_data = res.read(MAX_BYTES+1)
+        if len(raw_data) > MAX_BYTES:
```

Comment:
We should just check the content-length of the response and fail if it doesn't have one or is greater than this value.